src/react/InterfacePage.tsx

```diff
@@ -0,0 +1,259 @@
+/*
+   This file is part of Astarte.
+
+   Copyright 2020 Ispirata Srl
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import React, { useCallback, useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { Button, Container, Form, Row, Spinner } from 'react-bootstrap';
+import AstarteClient, { AstarteInterface } from 'astarte-client';
+
+import { useAlerts } from './AlertManager';
+import InterfaceEditor from './components/InterfaceEditor';
+import WaitForData from './components/WaitForData';
+import ConfirmModal from './components/modals/Confirm';
+import BackButton from './ui/BackButton';
+import useFetch from './hooks/useFetch';
+
+interface DeleteModalProps {
+  interfaceName: string;
+  interfaceMajor: number;
+  onCancel: () => void;
+  onConfirm: () => void;
+  isDeletingInterface: boolean;
+}
+
+const DeleteModal = ({
+  interfaceName,
+  interfaceMajor,
+  onCancel,
+  onConfirm,
+  isDeletingInterface,
+}: DeleteModalProps) => {
+  const [confirmString, setConfirmString] = useState('');
+
+  const canDelete = confirmString === interfaceName;
+
+  return (
+    <ConfirmModal
+      title="Confirmation Required"
+      confirmVariant="danger"
+      onCancel={onCancel}
+      onConfirm={onConfirm}
+      isConfirming={isDeletingInterface}
+      disabled={!canDelete}
+    >
+      <p>
+        You are going to remove{' '}
```

Comment:
i'd rather use `&nbsp;` instead of `{' '}`
or disable prettier for this sentence.